### ISSUE TITLE
Calendar: Uncaught Wrong time format using min/max date

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#4366]
From Stack Overflow: https://stackoverflow.com/questions/54025799/us-datetime-format-with-primefaces-6-2-issue

## 1) Environment
- PrimeFaces version: 6.3-SNAPSHOT
- Does it work on the newest released PrimeFaces version? NO
- Does it work on the newest sources in GitHub? NO
- Application server + version: ALL
- Affected browsers: ALL

## 2) Expected behavior

Calendar component using mindate/maxdate value will work as expected.

## 3) Actual behavior

JS Error is thrown "Uncaught Wrong time format"

## 4) Steps to reproduce

Create a calendar using pattern="MMM dd, yyyy hh:mm a" with a mindate value.

## 5) Sample XHTML

```xml
	<p:calendar id="datetime" value="#{testView.date1}" 
                            pattern="MMM dd, yyyy hh:mm a" 
                            showOn="button" 
                            showButtonPanel="true" 
                            mindate="#{testView.today}"
                            readonlyInput="true" timeZone="Asia/Karachi"/>
```

## 6) Sample bean

```java
public class TestView implements Serializable
{

    private Date date1;
    private Date today = new Date();

    @PostConstruct
    public void init()
    {
        today = new Date();
    }

    public Date getDate1() {
        return date1;
    }

    public void setDate1(Date date1) {
        this.date1 = date1;
    }

    public Date getToday() {
        return today;
    }

    public void setToday(Date today) {
        this.today = today;
    }

}
```

Test case: 
[pf-mindate.zip](https://github.com/primefaces/primefaces/files/2732822/pf-mindate.zip)

**Comment by pneumcke** [comment]
I have the same issue with 6.2.25 (and can not upgrade to 7.0 on that maintenance branch). The original error message from StackOverflow also reported 6.2.
Would it be possible to merge that fix to the 6.2 branch of PrimeFaces?
Thanks!

**Comment by melloware** [comment]
That is up to @mertsincan

**Comment by melloware** [comment]
@pneumcke If you need this fix right now just add this JS function to your page or library.

```javascript
PrimeFaces.widget.Calendar.prototype.configureTimePicker = function() {
        var pattern = this.cfg.dateFormat,
        timeSeparatorIndex = pattern.toLowerCase().indexOf('h');

        this.cfg.dateFormat = pattern.substring(0, timeSeparatorIndex - 1);
        this.cfg.timeFormat = pattern.substring(timeSeparatorIndex, pattern.length);

        //ampm
        if(this.cfg.timeFormat.indexOf('TT') != -1) {
            this.cfg.ampm = true;
        }
        
        // GitHub #4366 pass date and time settings for min/max date
        var timeSettings = {
                settings : this.cfg
        };
        var parseSettings = $.datepicker._getFormatConfig(timeSettings);

        //restraints
        if(this.cfg.minDate) {
            this.cfg.minDate = $.datepicker.parseDateTime(this.cfg.dateFormat, this.cfg.timeFormat, this.cfg.minDate, parseSettings, this.cfg);
        }

        if(this.cfg.maxDate) {
            this.cfg.maxDate = $.datepicker.parseDateTime(this.cfg.dateFormat, this.cfg.timeFormat, this.cfg.maxDate, parseSettings, this.cfg);
        }

        if(!this.cfg.showButtonPanel) {
            this.cfg.showButtonPanel = false;
        }

        if(this.cfg.controlType == 'custom' && this.cfg.timeControlObject) {
            this.cfg.controlType = this.cfg.timeControlObject;
        }

        if(this.cfg.showHour) {
            this.cfg.showHour = (this.cfg.showHour == "true") ? true : false;
        }

        if(this.cfg.showMinute) {
            this.cfg.showMinute = (this.cfg.showMinute == "true") ? true : false;
        }

        if(this.cfg.showSecond) {
            this.cfg.showSecond = (this.cfg.showSecond == "true") ? true : false;
        }

        if(this.cfg.showMillisec) {
            this.cfg.showMillisec = (this.cfg.showMillisec == "true") ? true : false;
        }
    };
```

**Comment by pneumcke** [comment]
Wow, thanks a lot for that quick fix! I'll see if I actually need a hotfix for the bug report.
Nevertheless, I would still prefer an "official" fix with a 6.2.26 version.
